src/components/views/messages/BridgeError.js

```diff
@@ -0,0 +1,139 @@
+            mxEvent.getId(),
+            "m.reference",
+            "de.nasnotfound.bridge_error",
+        );
+
+        return relations ? relations.getRelations() : [];
+    }
+
+    /**
+     * Returns a list of all users matched by the regex at the time of the event.
+     *
+     * @param {string} regexStr
+     * @returns {RoomMember[]}
+     */
+    _findMembersFromRegex(regexStr) {
+        const { room } = this.props;
+        const regex = new RegExp(regexStr);
+
+        // TODO[V02460@gmail.com]: Get room state at the proper point in time
+        const roomState = room.getLiveTimeline().getState(EventTimeline.FORWARDS);
+        const members = roomState.getMembers();
+
+        return members.filter(m => regex.test(m.userId));
+    }
+
+    /**
+     * Returns the network name and the affected users for the given relation.
+     *
+     * @param {MatrixEvent} relation
+     * @returns {{networkName: string, affectedUsers: RoomMember[]}}
+     */
+    _getRelationInfo(relation) {
+        if (!relation.event || !relation.event.content) {
+            return { networkName: "", affected_users: [] };
+        }
+        const content = relation.event.content;
+
+        const networkName = (typeof content.network_name === 'string' ?
+            content.network_name :
+            ""
+        );
+        const affectedUsersRegex = (Array.isArray(content.affected_users) ?
+            content.affected_users :
+            []
+        );
+        const affectedUsers = affectedUsersRegex.flatMap(
+            this._findMembersFromRegex.bind(this),
+        );
+
+        return { networkName, affectedUsers };
+    }
+
+    /**
+     * Returns the rendered string for the given relation.
+     *
+     * @param {{networkName: string, affectedUsers: RoomMember[]}} relationInfo
+     * @return {string}
+     */
+    _renderInfo(relationInfo) {
+        const usernames = relationInfo.affected_users.map(u => u.name);
+
+        if (relationInfo.networkName) {
+            return _t(
+                "%(networkName)s: %(affectedUsers)s",
+                {
+                    networkName: relationInfo.networkName,
+                    affectedUsers: usernames.join(", "),
+                },
+            );
+        } else {
+            return _t(
+                "Affected users: %(affectedUsers)s",
+                { affectedUsers: relationInfo.affectedUsers },
+            );
+        }
+    }
+
+    render() {
+        const relations = this._getRelations();
+        const isBridgeError = !!relations.length;
+
+        if (!isBridgeError) {
+            return null;
+        }
+
+        const relationInfos = relations.map(this._getRelationInfo.bind(this));
+        const renderedInfos = relationInfos.map(this._renderInfo.bind(this));
+
+        return (
+            <div className="mx_BridgeError">
+                ⚠ {_t("Event not delivered") + " (" + renderedInfos.join("; ") + ")" }
```

Comment:
I feel like this needs to be more clear that it didn't deliver it to the remote network, rather than failing to send outright or to remote matrix servers. I would say "Event not delivered to remote network", or "Event not delivered to $NETWORK".

Having this outside the brackets stops people from skipping over the important detail.

Reply:
also put the emoji in the translation please

Reply:
The warning icon should probably be a CSS background image, perhaps `img/warning.svg` masked with $warning-color. (see _EventTile.scss:299-316 as an example of how to do this).

Reply:
Did a first rework of the error messages, could you take another look?